code/datums/materials/basemats.dm

```diff
@@ -33,6 +33,16 @@
 	victim.apply_damage(10, BRUTE, BODY_ZONE_HEAD, wound_bonus = 5, sharpness = TRUE) //cronch
 	return TRUE
 
+/datum/material/glass/on_applied(atom/source, amount, material_flags)
+	. = ..()
+	if(isobj(source))
+		source.AddComponent(/datum/component/shatters_when_thrown, shard_type, round(amount / SHEET_MATERIAL_AMOUNT), SFX_SHATTER)
```

Comment:
This rounding makes no sense. If you throw a stack of glass containing 50 sheets you get like only 2 shards.

https://github.com/tgstation/tgstation/assets/110812394/113deefa-6f6b-472d-9fa0-8103bebc34e2

Also if you throw just one sheet of glass you get no shards. It just poofs

https://github.com/tgstation/tgstation/assets/110812394/21692144-d4fe-45e7-8c9b-4308ee1ec969

Also there is no random scatter effect for the 50 sheets breaking.

Reply:
the round function has failed me for the last time

Reply:
So I was right, there is something very wrong with the amount var that is doing something other than what I thought it'd do.

A stack of 50 glass sheets has an amount of 5000 (yes, 5000, not 50000 like it should be) whereas a chair has the correct amount (2000).

Reply:
Yes because that's how stuff breaks. Regardless of how messy it looks you can't explain how 48 sheets of glass vanished into thin air, 
Players should be able to collect the shards and recycle it to make the sheets again else this becomes a balance PR

Reply:
I don't think you quite understand the consequences of someone being able to create 50 glass shards with the click of a single button

Reply:
> I don't think you quite understand the consequences of someone being able to create 50 glass shards with the click of a single button

But if they don't spawn some shitter can just take all glass from protolathe and vanish it in the void.

Reply:
And it's just stupid that 50 stack of glass won't creat the same amount of shards. Like wtf

Reply:
> But if they don't spawn some shitter can just take all glass from protolathe and vanish it in the void.

Exactly,  these are the consequences.

Reply:
I've come up with an excellent solution.